File: Min_Max_Deletion.py
def solve(b):
    arr = b[:]
    score  = 0
    while len(arr)>1:
        pair = -1
        idx = -1
        for i in range(len(arr)-1):
            m = min(arr[i],arr[i+1])
            if m>pair:
                pair = m
                idx = i
        score += pair
        if arr[idx]>arr[idx+1]:
            arr.pop(idx)
        else:
            arr.pop(idx+1)
    print(score)
    
    
for _ in range(int(input())):
    n,q = map(int,input().split())
    a = list(map(int,input().split()))
    for _ in range(q):
        i,x = map(int,input().split())
        a[i-1] = x
        solve(a)
    
    

File: CodeChef/Contests/190/Min_Max_Deletion.java
import java.util.Scanner;

public class Min_Max_Deletion {
    public static void main(String[] args) throws java.lang.Exception {
        Scanner s = new Scanner(System.in);
        int t = s.nextInt();
        while (t-- > 0) {
            int n = s.nextInt(),q = s.nextInt();
            int[] arr = new int[n];
            for (int i=0;i<n;i++){
                arr[i] = s.nextInt();
            }
            long score = 0;
            for (int i=1;i<n;i++){
                score+=Math.min(arr[i],arr[i-1]);
            }
            while (q--!=0){
                int idx = s.nextInt()-1,x=s.nextInt();
                if (idx<n-1) score-=Math.min(arr[idx],arr[idx+1]);
                if (idx>0) score-=Math.min(arr[idx],arr[idx-1]);
                arr[idx] = x;
                if (idx<n-1) score+=Math.min(arr[idx],arr[idx+1]);
                if (idx>0) score+=Math.min(arr[idx],arr[idx-1]);
                System.out.println(score);
            }
            
        }
    }
}